tests/ImageSharp.Tests/Formats/Bmp/BmpDecoderTests.cs

```diff
@@ -39,22 +39,40 @@ public class BmpDecoderTests
         };
 
         [Theory]
-        [WithFileCollection(nameof(MiscBmpFiles), PixelTypes.Rgba32, false)]
-        [WithFileCollection(nameof(MiscBmpFiles), PixelTypes.Rgba32, true)]
-        public void BmpDecoder_CanDecode_MiscellaneousBitmaps<TPixel>(TestImageProvider<TPixel> provider, bool enforceDiscontiguousBuffers)
+        [WithFileCollection(nameof(MiscBmpFiles), PixelTypes.Rgba32)]
+        public void BmpDecoder_CanDecode_MiscellaneousBitmaps<TPixel>(TestImageProvider<TPixel> provider)
             where TPixel : unmanaged, IPixel<TPixel>
         {
+            using Image<TPixel> image = provider.GetImage(BmpDecoder);
+            image.DebugSave(provider);
+
+            if (TestEnvironment.IsWindows)
+            {
+                image.CompareToOriginal(provider);
+            }
+        }
+
+        [ActiveIssue("https://github.com/dotnet/arcade/issues/6393", TargetFrameworkMonikers.NetFramework)]
+        [Theory]
+        [WithFileCollection(nameof(MiscBmpFiles), PixelTypes.Rgba32)]
+        public void BmpDecoder_CanDecode_MiscellaneousBitmaps_WithLimitedAllocatorBufferCapacity<TPixel>(
+            TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            // dotnet xunit doesn't respect filter.
+            if (TestEnvironment.IsFramework)
```

Comment:
We are keeping  `dotnet xunit`  around only to do the x86 test run, which eventually happens only on Framework for some reason. Would be important to get the x86 coverage across all platforms with `dotnet test`, where `dotnet(.exe)` has to come from the 32 bit SDK, see https://github.com/xunit/xunit/issues/1123#issuecomment-449642224

Is this something we can easily get with `build-and-test.yml`?

Reply:
I don't know. Originally I wanted to do it via setup-dotnet but choosing the target framework there is not supported.

https://github.com/actions/setup-dotnet/issues/72

Reply:
I'm not sure the x86 SDKs are installed by default.

https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md#net-core-sdk